docs/reference/api-conventions.asciidoc

```diff
@@ -214,6 +214,34 @@ instead. When using this method, the `source_content_type` parameter
 should also be passed with a media type value that indicates the format
 of the source, such as `application/json`.
 
+[discrete]
+[[api-compatibility]]
+=== REST API version compatibility
+
+Major version upgrades often include a number of breaking changes 
+that impact how you interact with {es}. 
+While we recommend that you monitor the deprecation log and 
+update applications before upgrading {es},
+having to coordinate the necessary changes can be an impediment to upgrading. 
+
+To enable an application to continue to function without modification after an upgrade, 
```

Comment:
I think we should be more clear the functionality (semantics) are not necessarily maintained, the compatibility is with the syntax of rest request/response.